# Train/Test Pose Frame Statistics

`dataset/cropped_holistic_results_interpolated_split/{train,test}` 안의 각 `.npz` 샘플에서 `pose` 배열의 첫 번째 축을 프레임 수로 보고 평균을 계산합니다.


In [1]:
from pathlib import Path
import numpy as np
import pandas as pd

DATA_DIR = Path('/home/ubuntu/dataset/cropped_holistic_results_interpolated_split/train')
npz_paths = sorted(DATA_DIR.glob('*.npz'))
len(npz_paths)


In [2]:
rows = []
errors = []

for path in npz_paths:
    try:
        data = np.load(path, allow_pickle=False)
        pose = data['pose']
        rows.append({
            'sample': path.name,
            'pose_frames': int(pose.shape[0]),
            'pose_shape': tuple(int(x) for x in pose.shape),
        })
    except Exception as exc:
        errors.append({'sample': path.name, 'error': str(exc)})

df = pd.DataFrame(rows)
error_df = pd.DataFrame(errors)
df.head()


In [3]:
summary = {
    'num_samples': int(len(df)),
    'num_errors': int(len(error_df)),
    'mean_pose_frames': float(df['pose_frames'].mean()),
    'median_pose_frames': float(df['pose_frames'].median()),
    'min_pose_frames': int(df['pose_frames'].min()),
    'max_pose_frames': int(df['pose_frames'].max()),
    'std_pose_frames': float(df['pose_frames'].std(ddof=0)),
}
summary


{'num_samples': 962,
 'num_errors': 0,
 'mean_pose_frames': 59.21517671517672,
 'median_pose_frames': 54.0,
 'min_pose_frames': 13,
 'max_pose_frames': 151,
 'std_pose_frames': 21.483554083778664}

## Result

- 샘플 수: `962`
- 에러 수: `0`
- `pose` 프레임 수 평균: `59.21517671517672`
- 중앙값: `54.0`
- 최소/최대: `13` / `151`


In [ ]:
# 샘플별 pose 프레임 수 전체 보기
df.sort_values('sample').reset_index(drop=True)


In [ ]:
# 프레임 수가 가장 짧은/긴 샘플 확인
shortest = df.nsmallest(10, 'pose_frames')
longest = df.nlargest(10, 'pose_frames')
shortest, longest


## Test Split


In [4]:
TEST_DIR = Path('/home/ubuntu/dataset/cropped_holistic_results_interpolated_split/test')
test_npz_paths = sorted(TEST_DIR.glob('*.npz'))
len(test_npz_paths)


In [5]:
test_rows = []
test_errors = []

for path in test_npz_paths:
    try:
        data = np.load(path, allow_pickle=False)
        pose = data['pose']
        test_rows.append({
            'sample': path.name,
            'pose_frames': int(pose.shape[0]),
            'pose_shape': tuple(int(x) for x in pose.shape),
        })
    except Exception as exc:
        test_errors.append({'sample': path.name, 'error': str(exc)})

test_df = pd.DataFrame(test_rows)
test_error_df = pd.DataFrame(test_errors)
test_df.head()


In [6]:
test_summary = {
    'num_samples': int(len(test_df)),
    'num_errors': int(len(test_error_df)),
    'mean_pose_frames': float(test_df['pose_frames'].mean()),
    'median_pose_frames': float(test_df['pose_frames'].median()),
    'min_pose_frames': int(test_df['pose_frames'].min()),
    'max_pose_frames': int(test_df['pose_frames'].max()),
    'std_pose_frames': float(test_df['pose_frames'].std(ddof=0)),
}
test_summary


{'num_samples': 129,
 'num_errors': 0,
 'mean_pose_frames': 34.18604651162791,
 'median_pose_frames': 34.0,
 'min_pose_frames': 13,
 'max_pose_frames': 52,
 'std_pose_frames': 6.917865246138301}

## Test Result

- 샘플 수: `129`
- 에러 수: `0`
- `pose` 프레임 수 평균: `34.18604651162791`
- 중앙값: `34.0`
- 최소/최대: `13` / `52`


In [7]:
pd.DataFrame([summary, test_summary], index=['train', 'test'])
